### DISCUSSION TITLE
RBDL uses standard DH or modified DH dynamic parameters?

### DISCUSSION THREAD

**Discussion by Rivendharan-m** [rbdl/rbdl#91] [Q&A]
Hello everyone, 

I am student and I am currently working on a robotic project.  
Part of it requires the inverse dynamics of the robot (eg. ABB IRB 460). I am completely new to this field.

I find this RBDL GitHub code being suggested as a solution in online forum. 

Currently, I am trying to understand and develop the inputs required for the Inverse dynamics part of the code. 

I have few doubts/questions and looking for community help.

1. I am currently developing the URDF file for the robot.
 [https://robotics.stackexchange.com/questions/104763/ros-2-uses-modified-dh-notation-how-to-create-a-urdf-file-for-standard-dh-dyna]

with URDF, I have the above issue, if anyone knows solution please share it. 

2. In general, Do you have any suggestion for me, if I am going to  use this RBDL repo for an industrial robot with gearboxes , and also parallelogram structure [https://web.robodk.com/web?-OpenURL=//web.robodk.com/robotlib/lib/ABB-IRB-460.robot](url).

**note**: I know parallelogram structure create closed loop,but I am sure we can model this as open chain robot.

**Comment by ju6ge** [comment]
Hi pathavan,

I am sorry that it took so long for anybody to notice your question 🙈.  But it is nice of you to use the discussions feature of github instead of issues. It is where stuff like this belongs!

I hope you are already further down in your project. Sadly I don't really know how to help you with your question. So just a few thoughts from my side.

1. RBDL supports URDF Models, so that should no be a problem. Just make sure to enable the URDF-Addon!
2. Inverse Dynamics are supported by RBDL … I have not used it myself though. But you should find all you need in the documentation. Generally speaking you first need the motion of your robot in order to calculate the forces required to generate that motion. There are multiple ways to generate this motion:

- If you have a motion capture of the robot, you could use inverse kinematics to calculate the values of the joints
- Or you need to generate a motion using optimal control or other strategies.

3. Take a look at the examples directory of the code, there are different types of examples for simple problems. Hopefully they help you get an understanding of how to use RBDL.

Generally RBDL is a slow moving project, it is mostly developed and used by people doing Robotics Research. Most of them are not very active on Github because they are very busy people. I am not in the field of robotics anymore, but I do my best to at least keep the project alive as as Maintainer.
